ea_airflow_util/callables/ftp.py

```diff
@@ -28,15 +28,23 @@ def download_all(
     # Connect and download all selected files.
     hook = SFTPHook(ftp_conn_id)
 
-    files_to_download = [
-        file for file in hook.list_directory(remote_dir)
-        if not endswith or file.endswith(endswith)
-    ]
+    _, _has_extension = os.path.splitext(remote_dir)  # Overload to support files and directories.
```

Comment:
would it be safer or clearer to use `os.path.isfile()`?

Reply:
So we can't use that, because `isfile()` checks for the existence of the file on disk.